cypress/page/student-registration-form.page.ts

```diff
@@ -0,0 +1,83 @@
+import { data } from "cypress/types/jquery";
+import {PersonalInformation} from "../interfaces/PersonalInformation";
+class StudentRegistrationForm {
+  private nameText: string;
+  private lastNameText: string;
+  private emailText: string;
+  private genderBtn: string;
+  private mobileNumberText: string;
+  private hobbiesCheck: string;
+  private currentAddressText: string;
+  private submitBtn: string;
+  private stateList: string;
+  private cityList: string;
+  private NCROstate: string;
+  private delhiCity: string;
+  private hobbies: string;
+  private myhashhobbies: Map;
+
+  constructor() {
+    this.studentFormPageURL = 'https://demoqa.com/automation-practice-form';
+    this.bodyTr = 'tbody tr';
+    this.nameText = '#firstName'
+    this.lastNameText = '#lastName';
+    this.emailText = '#userEmail';
+    this.genderBtn = '[type="radio"]';
+    this.mobileNumberText = '#userNumber';
+    this.hobbiesCheck = '[type="checkbox"]';
+    this.currentAddressText = '#currentAddress';
+    this.submitBtn = '#submit';
+    this.stateList = '#state > div > div.css-1hwfws3 > div.css-1wa3eu0-placeholder';
+    this.cityList = '#city > div > div.css-1hwfws3 > div.css-1wa3eu0-placeholder';
+    this.NCROstate = "#react-select-3-option-0";
+    this.delhiCity = "#react-select-4-option-0";
+    this.hobbies = "Sports, Reading, Music";
+    this.myhashhobbies = new Map<string, string>([
+      ["1", "Sports"],
+      ["2", "Reading"],
+      ["3", "Music"]
+    ]);
+  }
+
+  public visitStudentForm(): void {
+    cy.visit(this.studentFormPageURL);
+  }
+
+  public fillForm(personalInformation: PersonalInformation): void {
```

Comment:
you are missing the dateOfBirth

Reply:
![image](https://user-images.githubusercontent.com/33491290/182277389-1ea258ab-7b48-4fb3-a2ac-f3a38a79adec.png)
![image](https://user-images.githubusercontent.com/33491290/182277417-0aaefda1-5eb4-4772-84c0-ef97746ee8ca.png)
![image](https://user-images.githubusercontent.com/33491290/182277449-5483038f-019b-412a-bf5c-76aca7381932.png)

when i clear the date text field the window stays blank

Reply:
yes, this is tricky, try to investigate how you can tell cypress to select all in order to you be able to write the date.